orc/src/main/java/org/apache/iceberg/orc/RemoveIds.java

```diff
@@ -46,6 +46,13 @@ public TypeDescription map(TypeDescription map, TypeDescription key, TypeDescrip
     return TypeDescription.createMap(key, value);
   }
 
+  @Override
+  public TypeDescription union(TypeDescription union, List<TypeDescription> options) {
```

Comment:
Could you clarify a little more on why this function is needed for this PR?

Reply:
We are leveraging the `RemoveIds` visitor to remove ids from a ORC schema, the issue is in LI-Iceberg we added union type support whereas the vanilla `RemoveIds` doesn't have the union case (because vanilla iceberg doesn't have union type), so we just need to add this override implementation to reconstruct the union schema case.